Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/Merge.swift

```diff
@@ -0,0 +1,131 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import ArgumentParser
+@_spi(FileManagerProtocol) import SwiftDocC
+import Foundation
+
+extension Docc {
+    /// Merge a list of documentation archives into a combined archive.
+    public struct Merge: ParsableCommand {
+        public init() {}
+        
+        public static var configuration = CommandConfiguration(
+            abstract: "Merge a list of documentation archives into a combined archive."
+        )
+        
+        private static let archivePathExtension = "doccarchive"
+        private static let catalogPathExtension = "docc"
+        
+        // The file manager used to validate the input and output directories.
+        //
+        // Provided as a static variable to allow for using a different file manager in unit tests.
+        static var _fileManager: FileManagerProtocol = FileManager.default
+        
+        // Note:
+        // The order of the option groups in this file is reflected in the 'docc merge --help' output.
+        
+        // MARK: - Inputs & outputs
+        
+        @OptionGroup(title: "Inputs & outputs")
```

Comment:
Is this a standard idiom for using `ParsableArguments` ? What's the reasoning behind having this option group? Why not simply place the properties directly on `Merge` and get rid of `InputAndOutputOptions` ?

Reply:
Defining option groups customizes how the help text for the command looks. It barely matters for a small command like this but for a command with many more options like `docc convert` it makes [a big difference][1]. I created an "Inputs & output" groups because two other commands already have one.

[1]: https://github.com/apple/swift-docc/pull/700#issuecomment-1684555034